### PR TITLE
Adds suggestions for monitoring and tuning the tail sampling processor to the README

### PR DESCRIPTION
**Pull request by swar8080**
**Description:** 
Hello, I had to read through the `tailsampling` processor source code to understand how to tune and monitor the component. Here's some possible documentation to help others do the same

### CODE REVIEW

File: processor/tailsamplingprocessor/README.md [hunk 1]
```diff
@@ -429,4 +429,83 @@ As a rule of thumb, if you want to add probabilistic sampling and...
...
+**Pre-emptively Preventing Dropped Traces**
+
+A trace will be dropped without sampling if it's removed from the circular buffer before `decision_wait`.
+
+To track how long traces remain in the buffer use:
+```
+otelcol_processor_tail_sampling_sampling_trace_removal_age
+```
+
+It may be useful to calculate latency percentiles like p1 and compare that value to `decision_wait`. Values close to `decision_wait` are at risk of being dropped if trace volume increases.
```
Comment (swar8080) [inline processor/tailsamplingprocessor/README.md:456]: Adding a new gauge tracking `decision_wait` could be helpful for alerts to avoid hard-coding the "close to `decision_wait`" part. Also for showing the value on a line graph of removal age percentiles

File: processor/tailsamplingprocessor/README.md [hunk 2]
```diff
@@ -429,4 +429,83 @@ As a rule of thumb, if you want to add probabilistic sampling and...
...
+
+**Slow Sampling Evaluation**
+```
+otelcol_processor_tail_sampling_sampling_decision_timer_latency
+```
+This measures latency of sampling a batch of traces and passing sampled traces through the remainder of the collector pipeline. A latency exceeding 1 second can delay sampling decisions beyond `decision_wait`, increasing the chance of traces being dropped before sampling.
```
Comment (swar8080) [inline processor/tailsamplingprocessor/README.md:462]: I didn't actually test this but i'm assuming that latency >1 second would cause some 1 second ticks from `time.Ticker` to be ignored. And maybe this is too much of an edge case to include?

File: processor/tailsamplingprocessor/README.md [hunk 3]
```diff
@@ -429,4 +429,83 @@ As a rule of thumb, if you want to add probabilistic sampling and...
...
\ No newline at end of file
+help resolve this error, at the expense of increased memory usage.
+
+## Monitoring and Tuning 
+
+See [metrics.go][metrics_go] for the full list metrics available for this component.
+
+### Dropped Traces
+
+A circular buffer is used to ensure the number of traces in-memory doesn't exceed `num_traces`. When a new trace arrives, the oldest trace is removed. This can cause a trace to be dropped before its sampled. To reduce the chance of this happening, either increase `num_traces` or decrease `decision_wait`. Both of those options increase memory usage.
```
Comment (tiffany76) [inline processor/tailsamplingprocessor/README.md:440]:
```suggestion
A circular buffer is used to ensure the number of traces in-memory doesn't exceed `num_traces`. When a new trace arrives, the oldest trace is removed. This can cause a trace to be dropped before it's sampled. To reduce the chance of this happening, either increase `num_traces` or decrease `decision_wait`. Both of those options increase memory usage.
```

File: processor/tailsamplingprocessor/README.md [hunk 4]
```diff
@@ -429,4 +429,83 @@ As a rule of thumb, if you want to add probabilistic sampling and...
...
+
+**Number of Traces Dropped**
+```
+otelcol_processor_tail_sampling_sampling_trace_dropped_too_early
+```
+
+**Pre-emptively Preventing Dropped Traces**
+
+A trace will be dropped without sampling if it's removed from the circular buffer before `decision_wait`.
```
Comment (tiffany76) [inline processor/tailsamplingprocessor/README.md:449]:
```suggestion
A trace is dropped without sampling if it's removed from the circular buffer before `decision_wait`.
```

File: processor/tailsamplingprocessor/README.md [hunk 5]
```diff
@@ -429,4 +429,83 @@ As a rule of thumb, if you want to add probabilistic sampling and...
...
+
+**Slow Sampling Evaluation**
+```
+otelcol_processor_tail_sampling_sampling_decision_timer_latency
+```
+This measures latency of sampling a batch of traces and passing sampled traces through the remainder of the collector pipeline. A latency exceeding 1 second can delay sampling decisions beyond `decision_wait`, increasing the chance of traces being dropped before sampling.
+
+It's therefore recommended to consume this component's output with components that trigger an asynchronous or fast export, like the `batch` processor.
+
+### Late Arriving Spans
```
Comment (tiffany76) [inline processor/tailsamplingprocessor/README.md:466]:
```suggestion
### Late-Arriving Spans
```

File: processor/tailsamplingprocessor/README.md [hunk 6]
```diff
@@ -429,4 +429,83 @@ As a rule of thumb, if you want to add probabilistic sampling and...
...
+**Sampling Policy Decision Frequency**
+
+To see how often each policy votes to sample a trace, use:
+
+```
+sum (otelcol_processor_tail_sampling_count_traces_sampled{sampled="true"}) by (policy) / 
+sum (otelcol_processor_tail_sampling_count_traces_sampled) by (policy) 
+```
+
+As a reminder, a policy voting to sample the trace does not guarantee sampling; An "inverted not" decision from another policy would still discard the trace.
```
Comment (tiffany76) [inline processor/tailsamplingprocessor/README.md:503]:
```suggestion
As a reminder, a policy voting to sample the trace does not guarantee sampling; an "inverted not" decision from another policy would still discard the trace.
```

**Review by tiffany76 — COMMENTED** [review]
Docs reviewer here. Just a few very minor copy edit suggestions. Thanks!

**Comment by swar8080** [comment]
> Docs reviewer here. Just a few very minor copy edit suggestions. Thanks!

Thanks @tiffany76

File: processor/tailsamplingprocessor/README.md (hunk 1 above)
Comment (jpkrohling) [reply processor/tailsamplingprocessor/README.md:456]: idea to implement as part of #31580

File: processor/tailsamplingprocessor/README.md (hunk 2 above)
Comment (jpkrohling) [reply processor/tailsamplingprocessor/README.md:462]: godoc for time.Ticker:

```
// NewTicker returns a new Ticker containing a channel that will send
// the current time on the channel after each tick. The period of the
// ticks is specified by the duration argument. The ticker will adjust
// the time interval or drop ticks to make up for slow receivers.
// The duration d must be greater than zero; if not, NewTicker will
// panic. Stop the ticker to release associated resources.
```

We don't do any async processing during the tick, so, I assume this is indeed the case.

File: processor/tailsamplingprocessor/README.md [hunk 7]
```diff
@@ -429,4 +429,83 @@ As a rule of thumb, if you want to add probabilistic sampling and...
...
+
+It may be useful to calculate latency percentiles like p1 and compare that value to `decision_wait`. Values close to `decision_wait` are at risk of being dropped if trace volume increases.
+
+**Slow Sampling Evaluation**
+```
+otelcol_processor_tail_sampling_sampling_decision_timer_latency
+```
+This measures latency of sampling a batch of traces and passing sampled traces through the remainder of the collector pipeline. A latency exceeding 1 second can delay sampling decisions beyond `decision_wait`, increasing the chance of traces being dropped before sampling.
+
+It's therefore recommended to consume this component's output with components that trigger an asynchronous or fast export, like the `batch` processor.
```
Comment (jpkrohling) [inline processor/tailsamplingprocessor/README.md:464]: I wonder if we should name the batch processor specifically, given we know we'll deprecate it in favor of native batching on exporters.

File: processor/tailsamplingprocessor/README.md [hunk 8]
```diff
@@ -429,4 +429,83 @@ As a rule of thumb, if you want to add probabilistic sampling and...
...
+
+### Late-Arriving Spans
+
+A span's arrival is considered "late" if it arrives after its trace's sampling decision is made. Late spans can cause different sampling decisions for different parts of the trace.
+
+There are two scenarios for late arriving spans:
+- Scenario 1: While the sampling decision of the trace remains in the circular buffer of `num_traces` length, the late spans inherit that decision. That means late spans do not influence the trace's sampling decision.
```
Comment (jpkrohling) [inline processor/tailsamplingprocessor/README.md:471]: Is this still the case? I thought I've seen that the latest code will just synchronously delete the entry from the entry map.

**Review by jpkrohling — COMMENTED** [review]
That's a great contribution, thank you!

File: processor/tailsamplingprocessor/README.md (hunk 7 above)
Comment (swar8080) [reply processor/tailsamplingprocessor/README.md]: good to know, it no longer mentions `batch` processor specifically, just fast/asynchronous consumers in general

File: processor/tailsamplingprocessor/README.md (hunk 8 above)
Comment (swar8080) [reply processor/tailsamplingprocessor/README.md:471]: I might be missing a code path but I only see deletion from `idToTrace` when a new span arrives and the circular buffer is full

image

And then i'm interpreting it as the late span hitting the below else block to pick-up the sampling decision. 

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/b1bbd587d4adc89f9cbc34754c4e67123d8e5283/processor/tailsamplingprocessor/processor.go#L426-L453

I'm just eye-balling all of this but happy to test this out to confirm

**Review by jpkrohling — APPROVED** [review]
